.github/workflows/test.yaml

```diff
@@ -58,16 +64,83 @@ jobs:
     - name: Build Go
       run: go build github.com/snapcore/snapd/...
     - name: Test C
-      run: |
-          cd ${{ github.workspace }}/src/github.com/snapcore/snapd/cmd/
-          make check
+      run: cd ${{ github.workspace }}/src/github.com/snapcore/snapd/cmd/ && make check
     - name: Test Go
-      run: go test github.com/snapcore/snapd/...
-  spread:
+      run: cd ${{ github.workspace }}/src/github.com/snapcore/snapd && ./run-checks --unit
+  spread-canary:
     needs: unit-tests
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
+      matrix:
+        system:
+        - ubuntu-16.04-64
+        - ubuntu-core-16-64
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Install spread
+      run: |
+          cd /tmp
+          curl -s -O https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz
+          tar xzvf spread-amd64.tar.gz
+          # Register a problem matcher to highlight spread failures
+          echo "::add-matcher::.github/spread-problem-matcher.json"
+    - name: Run spread tests
+      env:
+          SPREAD_GOOGLE_KEY: ${{ secrets.SPREAD_GOOGLE_KEY }}
+      run: /tmp/spread -abend google:${{ matrix.system }}:tests/...
+    - name: Discard spread workers
+      if: always()
+      run: |
+        shopt -s nullglob;
+        for r in .spread-reuse.*.yaml; do
+          /tmp/spread -discard -reuse-pid="$(echo "$r" | grep -o -E '[0-9]+')";
+        done
+  spread-stable:
+    needs: [unit-tests, spread-canary]
```

Comment:
make stable depends on the canary will take longer the whole execution, is it any reason for creating this new execution workflow?